turbopack/crates/turbopack-ecmascript/src/inlined_bytes_module.rs

```diff
@@ -0,0 +1,139 @@
+impl InlinedBytesJsModule {
+    #[turbo_tasks::function]
+    pub fn new(source: ResolvedVc<Box<dyn Source>>) -> Vc<Self> {
+        Self::cell(InlinedBytesJsModule { source })
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl Module for InlinedBytesJsModule {
+    #[turbo_tasks::function]
+    fn ident(&self) -> Vc<AssetIdent> {
+        self.source
+            .ident()
+            .with_modifier(rcstr!("static bytes in ecmascript"))
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl Asset for InlinedBytesJsModule {
+    #[turbo_tasks::function]
+    fn content(&self) -> Vc<AssetContent> {
+        self.source.content()
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl ChunkableModule for InlinedBytesJsModule {
+    #[turbo_tasks::function]
+    fn as_chunk_item(
+        self: ResolvedVc<Self>,
+        _module_graph: Vc<ModuleGraph>,
+        chunking_context: ResolvedVc<Box<dyn ChunkingContext>>,
+    ) -> Vc<Box<dyn turbopack_core::chunk::ChunkItem>> {
+        Vc::upcast(InlinedBytesJsChunkItem::cell(InlinedBytesJsChunkItem {
+            module: self,
+            chunking_context,
+        }))
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl EcmascriptChunkPlaceable for InlinedBytesJsModule {
+    #[turbo_tasks::function]
+    fn get_exports(&self) -> Vc<EcmascriptExports> {
+        EcmascriptExports::Value.cell()
+    }
+
+    #[turbo_tasks::function]
+    fn is_marked_as_side_effect_free(&self, _side_effect_free_packages: Vc<Glob>) -> Vc<bool> {
+        Vc::cell(true)
+    }
+}
+
+#[turbo_tasks::value]
+struct InlinedBytesJsChunkItem {
+    module: ResolvedVc<InlinedBytesJsModule>,
+    chunking_context: ResolvedVc<Box<dyn ChunkingContext>>,
+}
+
+#[turbo_tasks::value_impl]
+impl ChunkItem for InlinedBytesJsChunkItem {
+    #[turbo_tasks::function]
+    fn asset_ident(&self) -> Vc<AssetIdent> {
+        self.module.ident()
+    }
+
+    #[turbo_tasks::function]
+    fn chunking_context(&self) -> Vc<Box<dyn ChunkingContext>> {
+        *self.chunking_context
+    }
+
+    #[turbo_tasks::function]
+    async fn ty(&self) -> Result<Vc<Box<dyn ChunkType>>> {
+        Ok(Vc::upcast(
+            Vc::<EcmascriptChunkType>::default().resolve().await?,
+        ))
+    }
+
+    #[turbo_tasks::function]
+    fn module(&self) -> Vc<Box<dyn Module>> {
+        Vc::upcast(*self.module)
+    }
+}
+
+#[turbo_tasks::value_impl]
+impl EcmascriptChunkItem for InlinedBytesJsChunkItem {
+    #[turbo_tasks::function]
+    async fn content(&self) -> Result<Vc<EcmascriptChunkItemContent>> {
+        let content = self.module.content().file_content().await?;
+        match &*content {
+            FileContent::Content(data) => {
+                let encoded = data_encoding::BASE64
+                    .encode(&data.read().bytes().collect::<std::io::Result<Vec<u8>>>()?);
+                let inner_code = format!(
+                    "{TURBOPACK_EXPORT_VALUE}(Uint8Array.fromBase64({}));",
+                    StringifyJs(&encoded)
+                );
```

Comment:
Very minor nit:

Since `Uint8Array.fromBase64` defaults to a "loose" encoding that allows padding to be omitted (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Uint8Array/fromBase64#loose) we should omit padding, unless there's some performance reason to include the padding (doubtful?).


```suggestion
                let encoded = data_encoding::BASE64_NOPAD
                    .encode(&data.read().bytes().collect::<std::io::Result<Vec<u8>>>()?);
                let inner_code = format!(
                    "{TURBOPACK_EXPORT_VALUE}(Uint8Array.fromBase64({}));",
                    StringifyJs(&encoded)
                );
```

https://docs.rs/data-encoding/latest/data_encoding/constant.BASE64_NOPAD.html